### DISCUSSION TITLE
How can I override the home path during Repo() creation due to unsafe repository (REPO is owned by someone else)  error

### DISCUSSION THREAD

**Discussion by avatar-lavventura** [gitpython-developers/GitPython#1429] [Q&A] [answered]
I am using `git version 2.35.2`. I want to apply following solution [How can I specify custom global gitconfig path?](https://stackoverflow.com/questions/3592351/how-can-i-specify-custom-global-gitconfig-path) due to this error `fatal: unsafe repository (REPO is owned by someone else) with ubuntu 20.04 ` (Related: https://github.com/actions/checkout/issues/760).

> `HOME=/home/marco/.silly/ git commit -m "silly configuration"`

Is it possible to do same work around during `Repo` createtion, where I can override the home path?

The way I create my repo in python:
`repo = git.Repo(".", search_parent_directories=True)` here could I send `HOME` as an `env` variable?

**Comment by Byron** [comment, marked as answer]
It should be possible to change the environment of the python process itself, which should affect spawned commands as well.

I'd be surprised though that changing the home affects that user that git sees when checking it if that is the intend, I could be missing something though.

[This article](https://github.blog/2022-04-12-git-security-vulnerability-announced/) has a few suggestions on how to workaround this issue.

**Comment by avatar-lavventura** [comment]
@Byron When I run: `check_output(["env", f"HOME={home_dir}", "git", "status"])`, I was able to trick `git`, I have use solution here: https://stackoverflow.com/a/52227100/2402577. But I am not sure how this could be applied into `git.Repo()` call, might `os.environ['HOME'] = 'my_path'` be the solution.

**Reply by avatar-lavventura** [reply]
Article was the solution, where I had to add the safe.path but trick git to add it into `.gitconfig` of another user who owns the repo. I had to do:

```
check_output(["env", f"HOME={home_dir}", "git", "config", "--global", "--add", "safe.directory", path])
```

which could also be done easily using `GitPython`:

```
os.environ['HOME'] = 'my_path'  # this helped 
repo.config("--global", "--add", "safe.directory", path)
```
